src/app/components/blocks-editor/blocks-editor.component.ts

```diff
@@ -16,21 +16,40 @@ BLOCK_TYPES.forEach(element => _blockTypes[element.type]=element);
   templateUrl: './blocks-editor.component.html',
   styleUrls: ['./blocks-editor.component.scss']
 })
-export class BlocksEditorComponent implements OnInit {
+export class BlocksEditorComponent implements OnInit, OnChanges {
 
   @Input() blocks = [];
   @Input() blockIcons = _blockIcons;
   @Input() blockTypes = _blockTypes;
   @Output() blockUpdate = new EventEmitter();
 
+  blockTitles = [];
+
   constructor(
     private readonly dialog: MatDialog
   ) { }
 
-  ngOnInit() {
+  ngOnInit() {    
+  }
+
+  ngOnChanges(){
+    this.generateTitles();
+  }
+
+  generateTitles(){
+    this.blocks.forEach((block, i) => {            
+      const maxLength = 30       
+      let comment  = block.blockComment ? block.blockComment.split("\n")[0]: '';
+      // if the length is longer than max, truncate at word boundary
+      if (comment.length > maxLength) {
+        comment = comment.substr(0, comment.lastIndexOf(" ", maxLength)) + "...";
+      }      
+      this.blockTitles[i] = comment;
+    });
```

Comment:
Why not use it directly but style it with CSS?
https://developer.mozilla.org/en-US/docs/Web/CSS/text-overflow
This gives more control to the presentation layer that knows what the real limits are.
@the-greenman

Reply:
97% of browsers support it:
https://caniuse.com/text-overflow

Reply:
Suppose that would be fine.

Reply:
Now implemented with text-overflow. 
Still only getting the first line to allow for control of what appears.